### ISSUE TITLE
Contribution process aborted with assertion failed error

### ISSUE THREAD

**Issue by egri-nagy** [lightvector/KataGo#764]
KataGo was left running on an M1 Mac Mini, but after 11 games it aborted with:
```Assertion failed: (finishedGameData->endHist.moveHistory[i].pla == pla), function replayGameUpToMove, file play.cpp, line 2087.```

```gtp  version: KataGo v1.12.4
Git revision: 75280bf26582090dd4985dca62bc7124116c856d
Compile Time: Feb 21 2023 10:59:50
Using OpenCL backend
Compiled to support contributing to online distributed selfplay
```

**Comment by lightvector** [comment]
Thank you for reporting this, that's unexpected and definitely indicates something weird that I will try to investigate. Have you run KataGo on this system or on other systems before, and have you seen this error ever before?

**Comment by egri-nagy** [comment]
No, I haven't seen this error before. Contributing was flawless both on Mac and Linux computers before. Now for testing three new M2 Pro Mac Minis, I'm running KataGo in contribution mode on those, so I will let you know if it occurs again.
